Modules/EventHub/2.0/deploy.json

```diff
@@ -1,140 +1,324 @@
+                  "enabled": true
+                }
+              },
+              {
+                "category": "EventHubVNetConnectionEvent",
+                "enabled": true,
+                "retentionPolicy": {
+                  "days": "[parameters('logsRetentionInDays')]",
+                  "enabled": true
+                }
+              },
+              {
+                "category": "CustomerManagedKeyUserLogs",
+                "enabled": true,
+                "retentionPolicy": {
+                  "days": "[parameters('logsRetentionInDays')]",
+                  "enabled": true
+                }
               }
-            }
-          ]
+            ]
+          }
         }
       ]
     },
     {
       "type": "Microsoft.EventHub/namespaces/AuthorizationRules",
-      "name": "[concat(parameters('namespaceName'), '/', parameters('authorizationRulesRootManageSharedAccessKeyName'))]",
       "apiVersion": "2017-04-01",
+      "name": "[concat(parameters('namespaceName'),'/RootManageSharedAccessKey')]",
       "location": "[parameters('location')]",
-      "scale": null,
+      "dependsOn": [
+        "[variables('namespaceResourceId')]"
+      ],
       "properties": {
         "rights": [
           "Listen",
           "Manage",
           "Send"
         ]
-      },
+      }
+    },
+    {
+      "type": "Microsoft.EventHub/namespaces/AuthorizationRules",
+      "apiVersion": "2017-04-01",
+      "name": "[concat(parameters('namespaceName'),'/SendListenAccess')]",
+      "location": "[parameters('location')]",
       "dependsOn": [
-        "[resourceId('Microsoft.EventHub/namespaces', parameters('namespaceName'))]"
-      ]
+        "[variables('namespaceResourceId')]"
+      ],
+      "properties": {
+        "rights": [
+          "Listen",
+          "Send"
+        ]
+      }
+    },
+    {
+      "type": "Microsoft.EventHub/namespaces/eventhubs",
+      "apiVersion": "2017-04-01",
+      "name": "[concat(parameters('namespaceName'), '/',parameters('eventHubName'))]",
+      "location": "[parameters('location')]",
+      "dependsOn": [
+        "[variables('namespaceResourceId')]"
+      ],
+      "properties": {
+        "messageRetentionInDays": "[parameters('messageRetentionInDays')]",
+        "partitionCount": "[parameters('partitionCount')]",
+        "status": "Active"
+      }
+    },
+    {
+      "type": "Microsoft.EventHub/namespaces/eventhubs/consumergroups",
+      "apiVersion": "2017-04-01",
+      "name": "[concat(parameters('namespaceName'), '/',parameters('eventHubName'), '/$Default')]",
+      "location": "[parameters('location')]",
+      "dependsOn": [
+        "[resourceId('Microsoft.EventHub/namespaces/eventhubs', parameters('namespaceName'), parameters('eventHubName'))]",
+        "[variables('namespaceResourceId')]"
+      ],
+      "properties": {}
     }
   ],
   "outputs": {
-    "diagnosticEventHubName": {
+    "namespaceName": {
```

Comment:
I am in favor of this generic name, because EventsHub can be used for different purposes and not only diagnostics.

Reply:
Name changed